[IMP] runbot_travis2docker: Start ssh service

moylop260 opened the pull request:
- [x] Start ssh service
- [x] Add committer's github keys to `/home/odoo/.ssh/authorized_keys` (With `odoo:odoo` owner too)
- [x] Add tests to check
   - [x] if the file `/home/odoo/.ssh/authorized_keys` was updated from container
   - [x] if the field `build.docker_executed_commands` was updated
   - [x] if the `/etc/init.d/ssh status` is running

moylop260 committed:
[IMP] runbot_travis2docker: Start ssh service

moylop260 commented:
@JesusZapata 
Could you help me to finish this PR, please? (Check the main description)

Commit:
[IMP]  travis2docker: When the container is running it also runs the ssh server

JesusZapata commented:
@moylop260 
Only Just need to calculate the github user
You know how to find this information starting of repository?

moylop260 commented:
The json returned from [_get_pull_info](https://github.com/odoo/odoo-extra/blob/851320c0175/runbot/runbot.py#L504) method has user info
  - E.g. https://api.github.com/repos/odoo/odoo/pulls/1

For stable branches `refs/head/10.0` you could use a github api too (see the documentation)

For get the keys you can use: https://github.com/USER.keys
  - E.g. https://github.com/jesuszapata.keys


Note: You consider this change for gitlab too

Commit:
[IMP] runbot_travis2docker: Calculating the commit user

Commit:
[IMP] runbot_travis2docker: Adding test cases

JesusZapata commented:
@moylop260 
Thanks for you help.
I made the test for this funcionality

Commit:
[FIX] runbot_travis2docker: Change the order of the ssh_server tests

moylop260 commented on runbot_travis2docker/models/runbot_build.py:
```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
+        for build in self.browse(cr, uid, ids, context=context):
+            if (build.state == 'running' and build.job == 'job_30_run' and
+                    not build.docker_executed_commands):
+                session = requests.Session()
```
What about use `repo.github` method?

moylop260 commented on runbot_travis2docker/models/runbot_build.py:
```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
+                           build.docker_container, "curl",
+                           "https://github.com/%s.keys" %
+                           json['author']['login'], "-o",
+                           "/home/odoo/.ssh/authorized_keys"]
```
The original ones shouldn't be overwritten

moylop260 commented on runbot_travis2docker/models/runbot_build.py:
```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
+                           "/home/odoo/.ssh/authorized_keys"]
+                    _logger.info('Copy keys of github : ' + ' '.join(cmd))
+                    subprocess.call(cmd)
+                    cmd = ["docker", "exec", "--user=odoo",
```
This validation should be moved to tests section
IMHO here we are running `docker-exec` commands unnecessarily

moylop260 commented on runbot_travis2docker/tests/test_runbot_build.py:
```diff
@@ -78,6 +78,38 @@ def test_20_jobs_pr(self):
 
+    def test_30_ssh_server(self):
+        self.repo.update()
+        self.repo.killall()
```
These tests are too much expensive
Maybe we could re-use the original method to add custom validations for ssh keys
(I know we are breaking tests' standards)

moylop260 committed:
[REF] runbot_travis2docker: Minimal code and method inheritable to gitlab

moylop260 reviewed (commented):
@JesusZapata Could you check if is good for you my last commit?

moylop260 committed:
[REF] runbot_travis2docker: Add extra keys
Use extra authorized keys if you don't want allow ssh connection to host instead of just container builds

JesusZapata commented:
@moylop260 
I reviewed your change and all it is okay
Thanks for yours commits
